src/main/java/org/miracum/etl/fhirtoomop/mapper/EncounterInstitutionContactMapper.java

```diff
@@ -329,8 +329,13 @@ private VisitOccurrence createNewVisitOccurrence(
     var endDateTime = institutionContactOnset.getEndDateTime();
     var visitTypeConceptId = getVisitTypeConceptId(srcEncounter, endDateTime);
     var visitEndDateTime = setVisitEndDateTime(visitTypeConceptId, endDateTime, encounterId);
-    var visitSourceValue = cutString(encounterSourceIdentifier);
     var careSiteId = getCareSiteId(srcEncounter.getServiceProvider().getReferenceElement().getIdPart());
+    String visitSourceValue = null;
+    if(srcEncounter.getMeta().hasTag()){
+      visitSourceValue =  srcEncounter.getMeta().getTagFirstRep().getCode();
+    } else{
+      visitSourceValue = srcEncounter.getClass_().getDisplay();
+    }
     if (careSiteId == null){
```

Comment:
as Discussed with you It should be removed because in #8 you are passing `visitOccurrenceReason` to `createNewVisitOccurrence` and using it to set `visitSourcevalue`

Reply:
Okay. This can be removed after merging that PR.